### PR TITLE
[PayID] Add `PayIDUtil` and `PaymentPointer` 

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Add common PayID functionality:
- `PayIDUtil`: Static utilities for working with PayID
- `PaymentPointer`: Model object of a payment pointer

### Context of Change

This algorithm is modeled of Quilt Java code:
- Code: https://github.com/hyperledger/quilt/blob/master/spsp-parent/spsp-core/src/main/java/org/interledger/spsp/PaymentPointer.java
- Unit Tests: https://github.com/hyperledger/quilt/blob/master/spsp-parent/spsp-core/src/test/java/org/interledger/spsp/PaymentPointerTest.java

This is prep work to begin to make calls to a PayID service in higher level libraries (JS / Java / ILP). Since this code is somewhat complex I think it makes sense to place it into the common core to begin with and bind it upwards.

Based on the flux level of PayID / SDK features in general, I'm going to defer writing technical documentation for this at the moment. I think we'll do a single pass ahead of our next launch. I have updated CHANGELOG.md.

Payment pointers are generic across ILP / PayID so this will probably eventually be refactored into a `Common` folder. 

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After

New class introduced. 

## Test Plan

CI

### CODE REVIEW

File: src/PayID/pay-id-utils.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+   * Parse a payment pointer string to a payment pointer object
+   *
+   * @param paymentPointer The input string payment pointer.
+   * @returns A PaymentPointer object if the input was valid, otherwise undefined.
+   */
+  public static parsePaymentPointer(
+    paymentPointer: string,
+  ): PaymentPointer | undefined {
+    // Input must be ascii only.
+    if (!PayIDUtils.isASCII(paymentPointer)) return
```
[the remark this answers is not in the document]
Comment (keefertaylor) [reply src/PayID/pay-id-utils.ts]: Payment Pointers in Quilt specifically disallow non-ascii characters:
https://github.com/hyperledger/quilt/blob/master/spsp-parent/spsp-core/src/main/java/org/interledger/spsp/PaymentPointer.java#L91

And the test case: 
https://github.com/hyperledger/quilt/blob/master/spsp-parent/spsp-core/src/test/java/org/interledger/spsp/PaymentPointerTest.java#L78

I'd rather keep the behavior consistent but wrong than inconsistent and correct some of the times.

[the remark this answers is not in the document]
Comment (hbergren) [reply src/PayID/pay-id-utils.ts]: Can you add a TODO here at least to follow up about this? My understanding is that the Quilt implementation is incorrect.

[the remark this answers is not in the document]
Comment (keefertaylor) [reply src/PayID/pay-id-utils.ts]: Sure. I added an Asana task too so it gets brought up in the next Sprint at which time hopefully we can task and have a resolution.

**Comment by keefertaylor** [comment]
> Why does PayIDUtils exist here? What is the benefit of having PayIDUtils.parsePaymentPointer over having the PaymentPointer constructor be able to take a raw payment pointer string?

A couple reasons, most of which are person. 

First, I generally just prefer to keep validation logic out of classes. I get that we could have this logic embedded in constructors, but then your constructors get really long. 

Secondly, it's not like you're "making a class with inputs", it's more of a "conversion" which feels more "utility class"-ey to me. 

Thirdly, you can't overload constructors in JS. That means we'd need to make either a factory class or a static method if we wanted a constructor that took (1) host / path and (2) a payment pointer. When XpringClient had a static method for a constructor people often got confused and filed bugs, which makes me think this isn't a common pattern (happy to be told differently). 

You also can't fail constructors (and return undefined) in JS (AFAIK). This means that if you get erroneous input you have to throw. I'm not quite sure how I would handle exceptions being thrown in higher level languages (like Swift / Java) but I imagine it's a lot harder than simply returning a reference to undefined. 

I admit none of these options are particularly compelling on their own. If you have strong opinions or reasoning I'm happy to consider.

File: src/PayID/pay-id-utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (keefertaylor) [reply src/PayID/pay-id-utils.ts]: @hbergren: @sappenin  confirmed these are ascii only so I've reverted the todo and the implementation remains as is.

[the remark this answers is not in the document]
Comment (sappenin) [reply src/PayID/pay-id-utils.ts]: Yep, I'm forgetful today.

**Pull request merged**